### ISSUE TITLE
Disabling numbering in book class is really obnoxious

### ISSUE THREAD

**Issue by alerque** [simoncozens/sile#417]
When writing a document in `.sil` format controlling the numbering from the content makes some sense. `\chapter{...}` vs `\chapter[numbering=false]{...}` is reasonable. But when the content is coming  from some other format, styling a document is actually pretty hard the way it works right now. When chapters are coming from Markdown (`# ...`) it is possible to mark them as unnumbered in my Pandoc Writer (`# ... {.unnumbered}`) but doesn't even seem to be possible in the internal Markdown reader. In any event having this styling decision in the content is undesirable in the first place.

Not being able to pass an argument to the chapter command, I've taken to building my own chapter command just to handle stuff like this. In my case it keeps a global array of what levels get numbered, but for hack jobs like I just did for #413, you at least have in inject a value like this:

```Lua
local original_chapter = SILE.Commands["chapter"]
SILE.registerCommand("chapter", function (options, content)
  options.numbering = false
  original_chapter(options, content)
end, "Begin a new chapter")
```

That's a lot of shenanigans to default chapters to not have numbering. There must be a better way.

**Comment by Omikhleia** [comment]
One possible (?) way to look at this is to consider there are _two_ different things, and that the "numbering=xxx" is a bit misleading (as of SILE 0.11), mixing a bit of both things altogether:
- One is purely semantic : sectioning commands that do not increment counters and don't have a number, but behave exactly the same way otherwise. That's one would expect e.g. for "Prologue, Chapter 1, ..., Interlude., ... Chapter N, Epilogue". Or for sections (seen in some technical docs), "Prerequisites, Section 1, ..., Section N, Summary".
- Another is styling: how things show up (in the text flow, but also in TOC, etc.). That could include whether the numbers are shown or not (e.g. one could consider not showing numbers at all for say, subsubsections), but although plenty of other things (colors, skips, fonts, breaks...) which have very different ways of being done as of now (sometimes options, very often via _ad hoc_ hooks, etc.), or cannot be done without much redefinition (e.g. vertical skips, and... even twoside-page books with chapters not beginning on an odd page - I have seen a few of these). A more generalized attempt at defining styles might perhaps be possible. I am experimenting in that direction, maybe it will prove a false end, but just mentioning the rough idea here, as it somehow kind of relates to how sectioning command show up and how to configure it (and more generally, if I can make sense of it, the book-like class I'll come up with would somehow be heavy refactor of the default one ^^

**Comment by alerque** [comment]
At the moment I'm content with `SILE.setCommandDefaults()` for fixing this. It was introduced long after this issue in e8320d45 but directly addresses the problem of needing to always pass the same options and not wanting to modify the content to make it happen.

A further improvement *could* mean moving the default value of the numbering option to a *setting* and using the options as a setting override.

**Comment by Omikhleia** [comment]
This obviously doesn't contradict the other mentioned ways (such as `setCommandDefaults`) or alternative proposals for a "general" default value (such as using settings), but  as of Markdown:
> ...  (`# ... {.unnumbered}`) but doesn't even seem to be possible in the internal Markdown reader

Note that **lunamark** now supports header attributes (also expanding `-` to `.unnumbered`, so e.g. `# title {-}` also works transparently).

[markdown.sile](https://github.com/Omikhleia/markdown.sile) actually uses it (at this stage, still "vendoring" the updated lunamark code - since the latest improvements are not yet on luarocks - but the migration path when they eventually do would be logical).

**Comment by alerque** [comment]
Whether `{-}` or in SIL sources my view on this particular option is the same: setting it in content is useful for when a specific heading is different than others, but not for setting the default for typesetting purposes. For example my books (Markdown) use `{-}` for prefaces and appendixes but not for regular chapters. This goes for whether the book will be printed with numbered chapters or not. Even if the print edition doesn't use them, digital editions like epub might.

In other words content vs. presentation. Marking a chapter as unnumbered in content is different than disabling numbering in a specific output.

**Comment by Omikhleia** [comment]
> In other words content vs. presentation. 

I do fully concur (my own classes are based on styles to achieve this, but this is indeed the same core idea, content vs. actual presentation).
